src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -97,6 +101,36 @@ private static void setupLogger(Application application) {
 		application.addComponentMenuElement(Logger.class, "OFF logging", (ActionEvent e) -> logger.setLevel(Level.OFF));
 	}
 
+	/**
+	 * Feature for drawing lines with mouse. Left click = draw line, Right click = operateTo
+	 *
+	 * @param application Application context.
+	 */
+	private static void mouseClickConverter(Application application) {
+		JPanel jPanel = application.getFreePanel();
+		jPanel.addMouseListener(new MouseAdapter() {
+
+			@Override
+			public void mouseClicked(MouseEvent e) {
+				//Left click
+				if (SwingUtilities.isLeftMouseButton(e)) {
+					DriverFeature.getDriverManager()
+							.getCurrentDriver()
+							.operateTo(e.getX() - jPanel.getWidth() / 2, e.getY() - jPanel.getHeight() / 2);
+					System.out.println("New position x= " + e.getX());
+				}
+
+				//Right click
+				if (SwingUtilities.isRightMouseButton(e)) {
+					DriverFeature.getDriverManager()
+							.getCurrentDriver()
+							.setPosition(e.getX() - jPanel.getWidth() / 2, e.getY() - jPanel.getHeight() / 2);
+					System.out.println("Position saved x= " + e.getX());
+				}
+			}
+		});
+	}
+
```

Comment:
Too much changes in already existing class. Please extract this functionality to a separate class.